src/annoylib.h

```diff
@@ -313,14 +313,13 @@ struct Angular : Base {
   }
   template<typename S, typename T, typename Random>
   static inline void create_split(const vector<Node<S, T>*>& nodes, int f, size_t s, Random& random, Node<S, T>* n) {
-    Node<S, T>* p = (Node<S, T>*)malloc(s); // TODO: avoid
-    Node<S, T>* q = (Node<S, T>*)malloc(s); // TODO: avoid
+    char nodes_buffer[s * 2];
+    Node<S, T>* p = (Node<S, T>*)nodes_buffer;
+    Node<S, T>* q = (Node<S, T>*)nodes_buffer + 1;
```

Comment:
this is nice. however i think the second line has to be `    Node<S, T>* q = (Node<S, T>*)(nodes_buffer + s)`